File: app/src/main/java/com/example/benchmarks/models/AppComponent.java
package com.example.benchmarks.models;

import com.example.benchmarks.models.benchmark.BenchmarksModule;
import com.example.benchmarks.ui.benchmark.BenchMarkViewModelFactory;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = BenchmarksModule.class)
public interface AppComponent {
    void inject(BenchMarkViewModelFactory benchMarkViewModelFactory);
}

File: app/src/main/java/com/example/benchmarks/models/AppComponent.kt
package com.example.benchmarks.models

import com.example.benchmarks.ui.benchmark.BenchMarkViewModelFactory
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [BenchmarksModule::class])
interface AppComponent {
    fun inject(benchMarkViewModelFactory: BenchMarkViewModelFactory?)
}
